--- emesene-2.12.5/emesene/e3/common/pluginmanager.py~	2012-06-18 22:22:03.000000000 +0300
+++ emesene-2.12.5/emesene/e3/common/pluginmanager.py	2012-06-18 22:22:46.818149193 +0300
@@ -173,7 +173,7 @@
         for filename in os.listdir(dir_):
             path = os.path.join(dir_, filename)
             if filename.startswith(".") or \
-               not (os.path.isdir(path) or filename.endswith('.py')) or \
+               not (os.path.isdir(path) or filename.endswith('.py') or filename.endswith('.pyc')) or \
                filename in BLACKLIST:
                 continue
 
