--- emesene-2.11.11/emesene/pluginmanager.py~	2011-11-22 18:00:09.000000000 +0200
+++ emesene-2.11.11/emesene/pluginmanager.py	2011-11-22 18:02:42.495472765 +0200
@@ -173,7 +173,7 @@
         for filename in os.listdir(dir_):
             path = os.path.join(dir_, filename)
             if filename.startswith(".") or \
-               not (os.path.isdir(path) or filename.endswith('.py')) or \
+               not (os.path.isdir(path) or filename.endswith('.py') or filename.endswith('.pyc')) or \
                filename in BLACKLIST:
                 continue
 
